### PR TITLE
(feat)(Business Glossary) Glossary term relationship backend

### PR DESCRIPTION
**Pull request by Shikha-Trivedi-Saxo**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to #2864
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ */
+ @Aspect = {
+  "name": "glossaryRelatedTerms"
+}
+record GlossaryRelatedTerms {
+
+  /**
+   * Audit stamp containing who reported the lineage and when
+   */
+  auditStamp: AuditStamp
```
Comment (gabe-lyons) [inline metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl:17]: what will be the source of this audit stamp? we want to be conscious about including these only if we can populate them. will you have source data for created and updated at?

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryTermInfo.pdl [hunk 2]
```diff
@@ -47,4 +47,9 @@ record GlossaryTermInfo {
    * A key-value map to capture any other non-standardized properties for the glossary term
    */
   customProperties: map[string, string] = { }
+
+  /**
+   * Schema definition of the glossary term
+   */
+  schemaDefinition: optional string
```
Comment (gabe-lyons) [inline metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryTermInfo.pdl:54]: what is the difference between schemaDefinition and definition?

Comment (Shikha-Trivedi-Saxo) [reply metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryTermInfo.pdl:54]: schemaDefinition is the raw schema for Glossary terms whereas definition is the description for that term.

Comment (gabe-lyons) [reply metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryTermInfo.pdl:54]: would `rawSchema` make more sense as a title then?

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl (hunk 1 above)
Comment (Shikha-Trivedi-Saxo) [reply metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl]: Removed the auditStamp. We can have this when we want to store the version history of glossaryTerm relationships.

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryTermInfo.pdl (hunk 2 above)
Comment (Shikha-Trivedi-Saxo) [reply metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryTermInfo.pdl]: Renamed it to `rawSchema`

**Review by gabe-lyons — APPROVED** [review]
LGTM!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**